static/app/components/forms/form.tsx

```diff
@@ -235,21 +235,29 @@ function Form({
               )}
 
               <Observer>
-                {() => (
-                  <Button
-                    data-test-id="form-submit"
-                    priority={submitPriority ?? 'primary'}
-                    disabled={
-                      formModel.isError ||
-                      formModel.isSaving ||
-                      submitDisabled ||
-                      (requireChanges ? !formModel.formChanged : false)
-                    }
-                    type="submit"
-                  >
-                    {submitLabel ?? t('Save Changes')}
-                  </Button>
-                )}
+                {() => {
+                  return (
+                    <Button
+                      title={
+                        formModel.isInvalid || formModel.isError
+                          ? t('Required fields must be filled out')
```

Comment:
Is this always why a form would be invalid though?

Reply:
What if there's a regex for a text field that doesn't match or something?

Reply:
that makes sense. We can also allow custom labels later, but for now, how about 'Required fields must be filled out and contain valid inputs'?